cdap-ui/app/hydrator/services/create/stores/preview-store.js

```diff
@@ -41,6 +42,11 @@ var preview = (state = getInitialState(), action = {}) => {
       const {previewId} = action.payload;
       stateCopy.previewId = previewId;
       return Object.assign({}, state, stateCopy);
+    case previewActions.SET_RUNTIME_ARGUMENTS:
+      stateCopy = Object.assign({}, state);
+      const {runtimeArguments} = action.payload;
+      stateCopy.runtimeArguments = runtimeArguments;
+      return Object.assign({}, state, stateCopy);
```

Comment:
Also I am not sure why we are doing `Object.assign` twice. `Object.assign` in line 46 creates a shallow copy of state.

```
let runtimeArguments = action.payload.runtimeArguments;
return Object.assign({}, state, {runtimeArguments});
```
would be simpler right?